app/jobs/slack_poster_job.rb

```diff
@@ -9,13 +9,14 @@ class SlackPosterJob < ApplicationJob
   def perform(text, channel, options = {})
     raise "Invalid options, only #{VALID_OPTIONS.join(', ')} are permitted" unless valid_options?(options)
 
-    @webhook_url = ENV["SLACK_WEBHOOK_URL"]
-
-    post_to_slack(payload(text, channel, options)) if @webhook_url.present?
+    @webhook_url = ENV.fetch("SLACK_WEBHOOK_URL", nil)
```

Comment:
`ENV["SLACK_WEBHOOK_URL"]` should return nil if not present, unless you wanted to be explicit about the default return value.

```
irb(main):003> ENV["SLACK_WEBHOOK_URL"]
=> nil
```

Reply:
Oh good catch, thanks!